app/controllers/v1/providers_controller.rb

```diff
@@ -9,6 +9,7 @@ def update
         render "", status: :bad_request
       else
         provider.cookies_enabled = action == "accept"
+        provider.cookies_saved_at = Time.zone.now
         provider.save!
 
```

Comment:
Yours is fine, but it could be changed to a one-liner 
```suggestion
        provider.update!(cookies_enabled: action == "accept", cookies_saved_at:Time.zone.now)
```
replacing lines 11-13